//Test Script
/*
public class SetDifferenceCalculator {

    public Set<Integer> calculateSetDifference(Set<Integer> set1, Set<Integer> set2) {
        Set<Integer> differenceSet = new Set<Integer>(set1);
        differenceSet.removeAll(set2);
        return differenceSet;
    }
}
*/
PLACEHOLDER
public class CustomException extends Exception {}

try {
    SetDifferenceCalculator calculator = new SetDifferenceCalculator();
    
    Set<Integer> set1 = new Set<Integer>{1, 2, 3, 4, 5};
    Set<Integer> set2 = new Set<Integer>{3, 4, 5, 6};
    Set<Integer> resultSet = calculator.calculateSetDifference(set1, set2);
    Set<Integer> expectedResult = new Set<Integer>{1, 2};
    System.assertEquals(expectedResult,resultSet);
    
} catch (Exception exc) {
    throw new CustomException('There was an error: ' + exc.getMessage());
}
